### DISCUSSION TITLE
Warning: Failure in signing Android aab

### DISCUSSION THREAD

**Discussion by NoodleKidz** [airsdk/Adobe-Runtime-Support#2075] [Q&A]
Hi,

I have this message while compiling my package:

![failure-in-signing-android-aab](https://user-images.githubusercontent.com/91290636/185758008-9ddecf1c-61ef-44b9-83f9-1e6c098faaea.jpg)

I thought it was OK, but when I sent it to my Android phone and tried to install it, it says "There was a problem parsing the package".

I use Adobe Animate CC 2022 22.0.7.

I didn't encounter this problem with Adobe Animate CC 2019, but 2019 version cannot compile 64bit.

Please help. 

Thank you

**Comment by NoodleKidz** [comment]
Also, I looked up some possible answers and I tried modifying the adt.cfg and put the path of the SDK to where the SDK folder is, and it gives me this error;

![error-compile](https://user-images.githubusercontent.com/91290636/185765236-bdd86c73-1767-496f-8082-b979ded92a25.jpg)

Please help, thank you.

**Reply by ajwfrost** [reply]
Hi

Normally, you won't need to sign an AAB file to start with, since when you turn it into APKs (so that you can put it onto your device) you provide the signing information there. So it may be that part that's failing .. Just checking though, if you're using Animate and ending up with an AAB file, are you using the configuration file to override the APK build and turn it into an AAB build?

The failure in signing is most likely due to either a Java compatibility issue or a missing/old Android SDK . If you update the adt.cfg file to un-comment the "DebugOut" line, please try it again and then find the log file "c:\users\antony\adt.log" which should have more information in it.

The error about platform directories looks like you're giving it an AIR SDK folder - but that variable is actually where it's looking for the Android SDK. In order to create AAB files we rely on the Android tools, so please make sure you've got a recent version of Android Studio installed and an appropriate Android SDK (by default in c:\users\antony\appdata\local\android\sdk).

thanks

**Reply by NoodleKidz** [reply]
Thank you for the reply, now it compiles without error (see figure 1 below), however, when I try to install it on my phone or on android emulator, it shows me "There was a problem parsing a package" (see figure 2 below), does it mean I cannot upload this to Google Play and users cannot download and install the app?

Thank you again for your help.

![APK-created-succesfully](https://user-images.githubusercontent.com/91290636/185797923-a479bd70-ba6c-487e-aaff-ee110b24c756.jpg)
![Screenshot_20220821-110948](https://user-images.githubusercontent.com/91290636/185798047-67c0fb0f-57e2-4ed3-84a7-d9b213d63487.png)

**Reply by ajwfrost** [reply]
Are you trying to install an AAB file by copying it onto the device and opening it from there? I don't think that would work, I think that devices have to work with APK files which you can create from an AAB using the "bundletool" which comes in the Android tools (or you can install via the AIR ADT utility which uses bundletool to create and then install the APKs).

But also - looking at the screenshot, it's very suspicious for an AAB file to be the same size as an APK! An AAB should contain support for all the platforms and would be a lot larger.. so something may not be quite right with the files... worth opening it up in a zip manager (or in Android Studio) and trying to see whether it's actually the right file format..

thanks

**Reply by NoodleKidz** [reply]
actually, animate compiles it as APK, I copied it and renamed it as AAB

**Reply by ajwfrost** [reply]
Okay - yes a restriction in Animate at the moment. Why did you want an AAB though? that's mostly I think just for uploading to the Play Store...

If you need to create an AAB using Animate, you need to go through a bit of a hack... you would need to edit the config file "adt.cfg" that's in the AIR SDK "lib" folder, and add a line `CreateAndroidAppBundle=true`. Then if you re-do the Animate build, it will look like it's given you an APK file but this one will actually be an AAB file (and should be a lot larger!).

thanks

**Reply by NoodleKidz** [reply]
yes,  Play Store no longer accept APK, that's why.

In the past, everytime I compiled into APK, I tried to play it on my phone first to find bugs, but now since I compiled into 64bit, renamed the APK into AAB, I can't even install in on my phone anymore, thus no way for me to play it (on my phone).

This is my config, if it helps;

![adt](https://user-images.githubusercontent.com/91290636/185816418-43a44f9e-9e5c-4fb8-b20d-908823f0f424.jpg)

Is it normal that once Adobe Animate compiles it in 64bit, we can no longer install it manually on our phones anymore?

If we rename that APK into AAB and upload it to Google Play console, it will work just fine?

Thank you

**Reply by ajwfrost** [reply]
With the above config, you should end up with an AAB file that you can then upload to the Play console, because you have that `CreateAndroidAppBundle` set to true. So whatever comes out of that should be fairly large and if you open it within a zip manager, you'll see a file `BundleConfig.pb` at the top level. This file can't be directly installed onto a phone, you can use `bundletool` or you can run `adt -installApp -platform android -package outputfile.aab` (it would have to be named .aab to work, as we check the extension and then do the installation via `bundletool`...)

If you don't have that `CreateAndroidAppBundle` setting, then Animate would give you an APK file (with armv8 binaries due to your `OverrideArch` setting) - if you open this in a zip manager, you'll see `AndroidManifest.xml` at the top level. This file can be directly installed onto your phone via ADB, ADT (or Animate), or just copying to mass storage and launching from the phone's file manager. Typically I would suggest people do this during all development/testing/debugging, and then switch over to create an app bundle just for uploading to the Play console. If you then use the Play store to install this on your test devices, Google will just give your phones the appropriate APKs that it pulls out of the bundle file..

thanks

**Comment by NoodleKidz** [comment]
Ah, got it, that explains everything.

Thank you so much for the explanation and your patience.
